Message user when a job succeeds after 24 hours

nanotone opened the pull request:
Connects #12670 

### Description
Message the user when a job succeeds more than 24 hours after it was submitted.

### Acceptance Criteria
- [ ] When an asyncable job gets successfully processed more than 24 hours after it was submitted, automatically send the user a message to their inbox.
- [ ] Include a link to the job detail page where the user can see when the job got processed and the notes on that job.

### Testing Plan
1. Create an HLR or SC, submitted more than 24 hours in the past (use a factory or existing test to help, see item 5 or 6 below)
2. `pry> hlr.processed!`
3. Load `/inbox` as the user who owns this job
4. Verify that a success message landed in the inbox.
5. Added test for job messaging service class
6. Added inbox feature test

### User Facing Changes
<img width="968" alt="Screen Shot 2019-11-25 at 10 13 16" src="https://user-images.githubusercontent.com/282869/69552348-4cf17a80-0f6c-11ea-8d26-bd582ca7abb7.png">
Sample job success message

### Notes
This mirrors the implementation of #12767, which overrides an `Asyncable` lifecycle method in `ClaimReview` to also trigger a call to the job messaging service class.

pkarman commented on app/services/asyncable_job_messaging.rb:
```diff
@@ -42,6 +42,21 @@ def handle_job_failure
+    return if job.messages.failing_job_succeeded.any?
+
+    message_text = <<-EOS.strip_heredoc
+      <a href="#{job.path}">#{job.class} #{job.id}</a> has successfully been processed. No further action is necessary. If you have opened a support ticket for this issue, you may inform them that it may be closed.
```
since this is a HEREDOC, might as well introduce some newlines to make it more readable?

pkarman commented on spec/services/asyncable_job_messaging_spec.rb:
```diff
@@ -80,4 +80,34 @@
+
+    context "when a success happens after 24 hours of failure" do
+      it "sends a message to the job owner" do
+        job
```
add a `let!(:job)` above?

pkarman commented on spec/services/asyncable_job_messaging_spec.rb:
```diff
@@ -80,4 +80,34 @@
+      it "sends a message to the job owner" do
+        job
+        messaging.handle_job_failure
+        Timecop.travel(Time.zone.now.tomorrow)
```
timecop.travel in a block?

nanotone commented:
@pkarman Addressed all your comments; it was all stuff I did in parallel with the failure-message PR, that I forgot to update during a rebase 🤦‍♂ -- ready for another look :bowtie:
